src/crab/fwd_analyzer.cpp

```diff
@@ -123,17 +123,16 @@ std::pair<invariant_table_t, invariant_table_t> run_forward_analyzer(cfg_t& cfg,
     // Go over the CFG in weak topological order (accounting for loops).
     interleaved_fwd_fixpoint_iterator_t analyzer(cfg);
     if (thread_local_options.check_termination) {
-        std::vector<label_t> cycle_heads;
-        for (auto& component : analyzer._wto) {
-            if (const auto pc = std::get_if<std::shared_ptr<wto_cycle_t>>(&component)) {
-                cycle_heads.push_back((*pc)->head());
+        // For every potential loop header, initialize the loop counter to 0.
+        for (const auto& label : cfg.potential_loop_headers()) {
+            // Skip non-deterministic blocks. Only deterministic blocks have loop counters.
```

Comment:
I'm not sure what does "deterministic block" mean.

Reply:
Hoping the new comment is clearer.

There are two classes of labels:
1) Labels added from the original assembly.
2) Labels added from the non-determination path.

Only labels of the first class will have the loop counters added.